core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -586,6 +612,16 @@ protected void writeSnapshotList(List<SnapshotId> snapshots) throws IOException
      * @throws IOException I/O errors
      */
     protected List<SnapshotId> readSnapshotList() throws IOException {
+        try {
+            if (snapshotsBlobContainer.blobExists(SNAPSHOTS_FILE) == false) {
```

Comment:
First, we need to check if the repository has a snapshot index file.  If it is a fresh repository, it won't have that snapshot index file and the `readBlob` call just below will throw exceptions.